extensions/typescript-language-features/src/languageFeatures/definitions.ts

```diff
@@ -29,7 +30,16 @@ export default class TypeScriptDefinitionProvider extends DefinitionProviderBase
 		}
 
 		const span = response.body.textSpan ? typeConverters.Range.fromTextSpan(response.body.textSpan) : undefined;
-		return response.body.definitions
+		let definitions = response.body.definitions;
+
+		if (vscode.workspace.getConfiguration(document.languageId).get('preferGoToSourceDefinition', false) && this.client.apiVersion.gte(API.v470)) {
```

Comment:
If this option is enabled, can we avoid the `definitionAndBoundSpan` call? Is there some other way to get the `span`?

Reply:
That’s what my PR description discusses a bit. I’m interested to get your thoughts.